tests/server/test_build_logger.py

```diff
@@ -0,0 +1,154 @@
+        - target: engine_outputs
+          mappings:
+           - func: np.mean
+             frequency: 4"""  # noqa E501
+
+PIPELINE_CONFIG_1 = """
+- name: question_answering
+  targets:
+    - target: pipeline_outputs
+      mappings:
+       - func: builtins:identity
+         frequency: 2
+       - func: tests/deepsparse/loggers/test_data/metric_functions.py:user_defined_identity 
+         frequency: 3
+    - target: engine_outputs
+      mappings:
+       - func: np.mean
+         frequency: 4"""  # noqa E501
+
+LOGGER_3 = logger_objects.FunctionLogger(
+    logger=logger_objects.PythonLogger(),
+    config=MultiplePipelinesLoggingConfig(
+        pipelines=[
+            PipelineLoggingConfig(**pipeline_config)
+            for pipeline_config in yaml.safe_load(PIPELINE_CONFIG_1)
+        ]
+    ),
+)
+
+YAML_CONFIG_4 = """loggers:
+    - python
+endpoints:
+    - task: question_answering
+      route: /unpruned/predict
+      model: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/base-none
+      batch_size: 1
+      data_logging:
+        - target: pipeline_outputs
+          mappings:
+           - func: builtins:identity
+             frequency: 2
+           - func: tests/deepsparse/loggers/test_data/metric_functions.py:user_defined_identity
+             frequency: 3
+    - task: question_answering
+      route: /unpruned/predict
+      model: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/base-none
+      batch_size: 1
+      name: question_answering_
+      data_logging:
+        - target: engine_outputs
+          mappings:
+           - func: np.mean
+             frequency: 4"""  # noqa E501
+
+PIPELINE_CONFIG_2 = """- name: question_answering
+  targets:
+    - target: pipeline_outputs
+      mappings:
+       - func: builtins:identity
+         frequency: 2
+       - func: tests/deepsparse/loggers/test_data/metric_functions.py:user_defined_identity
+         frequency: 3
+- name: question_answering_
+  targets:
+    - target: engine_outputs
+      mappings:
+       - func: np.mean
+         frequency: 4"""  # noqa E501
+
+LOGGER_4 = logger_objects.FunctionLogger(
+    logger=logger_objects.PythonLogger(),
+    config=MultiplePipelinesLoggingConfig(
+        pipelines=[
+            PipelineLoggingConfig(**pipeline_config)
+            for pipeline_config in yaml.safe_load(PIPELINE_CONFIG_2)
+        ]
+    ),
+)
+
+
+@pytest.mark.parametrize(
+    "yaml_server_config,expected_logger",
+    [
+        (YAML_CONFIG_1, LOGGER_1),
+        (YAML_CONFIG_2, LOGGER_2),
+        (YAML_CONFIG_3, LOGGER_3),
+        (YAML_CONFIG_4, LOGGER_4),
+    ],
+)
+def test_build_logger(yaml_server_config, expected_logger):
+    obj = yaml.safe_load(yaml_server_config)
+    server_config = ServerConfig(**obj)
+    logger = build_logger(server_config)
+    _is_equal(logger, expected_logger)
+
+
+def _is_equal(logger, expected_logger):
+    # figure out a smart way to test here
+    assert True
```

Comment:
For now, I am testing it by pickling the class objects and comparing their pickled definitions. Once this PR gets approved, I will write a more appropriate code for comparison between `logger` and `expected logger`